### ISSUE TITLE
Can't push to gcp because it tries to fetch auth. method from our landing page server which results in a 404.

### ISSUE THREAD

**Issue by TimVonsee** [GoogleContainerTools/jib#3322]
**Environment**:

- *Jib version: 3.1.1* 
- *Build tool: Maven 2.81* 

**Description of the issue**:
I can't push to gcp because its not picking up that I want to use gcloud authentication to push to our gcp container registry. It tries to fetch an auth. method from our homepage example.com (it probably resolved this url by our reversed packaging?).
I can manually push images to gcp, I have setup docker and [Jib + gcloud authentication](https://github.com/GoogleContainerTools/jib/blob/master/docs/configure-gcp-credentials.md. But every time I run `mvn compile jib:build` the build fails due to the fact that it is trying to "fetch retrieve authentication method for example.com" (see log below). Is there a way to force it to just use `docker-credential-gcr`?

**`jib-maven-plugin` Configuration**: 
```xml
# parent pom
<plugin>
          <groupId>com.google.cloud.tools</groupId>
          <artifactId>jib-maven-plugin</artifactId>
          <version>3.1.1</version>
          <configuration>
            <from>
              <image>openjdk:${jdk.version}-jre-slim-buster</image>
            </from>
            <to>
              <tags>
                <tag>latest</tag>
                <tag>${project.version}</tag>
              </tags>
            </to>
          </configuration>
</plugin>

# child pom
<build>
    <plugins>
      <plugin>
        <groupId>com.google.cloud.tools</groupId>
        <artifactId>jib-maven-plugin</artifactId>
        <configuration>
          <to>
            <image>gcr.io/project/example</image>
          </to>
          <container>
            <mainClass>com.example.app.MainKt</mainClass>
          </container>
        </configuration>
      </plugin>
    </plugins>
  </build>
```

**Log output**: 
```
[INFO] Building classes layer...
[INFO] Building jvm arg files layer...
[WARNING] The credential helper (docker-credential-desktop) has nothing for server URL: example.com

Got output:

credentials not found in native keychain

[INFO] The base image requires auth. Trying again for openjdk:11.0.11-9-jre-slim-buster...
[WARNING] The credential helper (docker-credential-desktop) has nothing for server URL: registry-1.docker.io

Got output:

credentials not found in native keychain

[WARNING] The credential helper (docker-credential-desktop) has nothing for server URL: registry.hub.docker.com

Got output:

credentials not found in native keychain

[WARNING] The credential helper (docker-credential-desktop) has nothing for server URL: index.docker.io

Got output:

credentials not found in native keychain

[WARNING] The credential helper (docker-credential-desktop) has nothing for server URL: docker.io

Got output:

credentials not found in native keychain

[INFO] Using base image with digest: sha256:550345cd237e42a5772b8eb9c76cb7020c62645de5da77649f821e3c556f
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  5.117 s
[INFO] Finished at: 2021-06-25T15:37:34+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:3.1.1:build (default-cli) on project app: Tried to retrieve authentication method for com.example but failed because: registry returned error code 404; possible causes include invalid or wrong reference. Actual error output follows:
[ERROR] <!DOCTYPE html><html class="no-js" lang="en"><head>
[ERROR] <meta charset="utf-8">
[ERROR] <meta http-equiv="Content-Security-Policy" content="">
[ERROR] <meta http-equiv="x-ua-compatible" content="ie=edge">
...
```

**Comment by dzou** [comment]
From the logs you can see it that it is getting stuck trying to download the `openjdk:${jdk.version}-jre-slim-buster` base docker image; it is not related to Google auth but authenticating to a Docker container registry. Also note, Jib tries to download this image from several places -- first com.example, then registry.hub.docker.com, then index.docker.io, etc. It does not simply fail at `example.com`.

So it's basically saying that you have not configured any credentials for downloading docker images from any of these sources.

I think this is similar to the problem here:
https://github.com/GoogleContainerTools/jib/issues/2182#issuecomment-561761583

Basically you have to ensure the `docker-credential-gcr` also has credentials for one of the docker registries that you will download your base image from.

**Comment by TimVonsee** [comment]
I see, now I get it. Thanks for the quick reply!

**Comment by dzou** [comment]
Hi @TimVonsee -- Looking at the logs, I think I may be mistaken. I think the problem is not related to the base image actually.

According to @chanseokoh, it looks like Jib is told to contact `example.com` based on `jib.to.image`. So somewhere in your config, `jib.to.image` is set to `example.com/...`.

If you can give it another try on the command line, try overriding the `jib.to.image` manually and see if that fixes the problem:

```
mvn -Djib.to.image=gcr.io/your-project/image-name jib:build
```

If that works, then maybe there is some config not being being overriden for the `jib.to.image`.

**Comment by TimVonsee** [comment]
I overlooked the fact that the in the parent pom's `<properties>` the image tag was set to `<image>com.example/${project.artifactId}</image>`. Because of this, the override didn't work either and made me re-check our parent POM's config, as you pointed out, and found the error. Anyway, thanks for the help!

**Comment by dzou** [comment]
Nice, glad to hear you figured it out :+1: .

Going to close, feel free to re-open if you have other questions.
